oktapam/client/secret_folder.go

```diff
@@ -0,0 +1,309 @@
+	var list []SecretFolder
+	request := sdkCLient.SDKClient.SecretsAPI.ListTopLevelSecretFoldersForProject(ctx, resourceGroupID, projectID, sdkCLient.Team)
+	resp, httpResp, callErr := request.Execute()
+	if httpResp != nil {
+		if _, err := checkStatusCodeFromSDK(httpResp, 200); err != nil {
+			return nil, err
+		}
+	} else if callErr != nil {
+		return nil, callErr
+	}
+
+	for _, s := range resp.List {
+		if s.Type == pam.SecretType_FOLDER {
+			list = append(list, ConvertSecretOrFolderListResponseToSecretFolder(s))
+		}
+	}
+
+	return list, nil
+}
+
+func ListSecretFoldersUnderPath(ctx context.Context, sdkClient SDKClientWrapper, resourceGroupID string, projectID string, path string) ([]SecretFolder, error) {
+	parentFolder, err := ResolveSecretFolder(ctx, sdkClient, resourceGroupID, projectID, path)
+	if err != nil {
+		return nil, err
+	}
+
+	var list []SecretFolder
+	request := sdkClient.SDKClient.SecretsAPI.ListSecretFolderItems(ctx, sdkClient.Team, resourceGroupID, projectID, *parentFolder.ID)
+
+	for {
+		resp, httpResp, callErr := request.Execute()
+		if httpResp != nil {
+			if _, err := checkStatusCodeFromSDK(httpResp, 200); err != nil {
+				return nil, err
+			}
+		} else if callErr != nil {
+			return nil, callErr
+		}
+
+		for _, s := range resp.List {
+			if s.Type == pam.SecretType_FOLDER {
+				list = append(list, ConvertSecretOrFolderListResponseToSecretFolder(s))
+			}
+		}
+
+		linkHeader := httpResp.Header.Get("Link")
+		if linkHeader == "" {
+			break
+		}
+
+		links := linkheader.Parse(linkHeader)
+
+		var nextURLStr string
+		for _, link := range links {
+			if link.Rel == "next" {
+				nextURLStr = link.URL
+			}
+		}
+
+		if nextURLStr == "" {
+			break
+		}
+
+		nextURL, err := url.Parse(nextURLStr)
+		if err != nil {
+			return nil, err
+		}
+		values := nextURL.Query()
+		offset, ok := values["offset"]
+		if !ok || len(offset) != 1 {
+			return nil, fmt.Errorf("invalid offset")
+		}
+
+		request = request.Offset(offset[0])
+	}
+
+	return list, nil
+}
+
+func ConvertSecretOrFolderListResponseToSecretFolder(resp pam.SecretOrFolderListResponse) SecretFolder {
+	return SecretFolder{
+		Name:            &resp.Name,
+		ID:              &resp.Id,
+		Description:     resp.Description.Get(),
+		ResourceGroupID: resp.ResourceGroup.Id,
+		ProjectID:       resp.Project.Id,
+	}
+}
+
+func ConvertResolveSecretOrFolderResponse(resp *pam.ResolveSecretOrFolderResponse) SecretFolder {
+	var parentSecretFolderID *string
+
+	if len(resp.Path) != 0 {
+		parentSecretFolderID = &resp.Path[len(resp.Path)-1].Id
+	}
+
+	return SecretFolder{
+		Name:            &resp.Name,
+		ID:              &resp.Id,
+		Description:     resp.Description.Get(),
+		ResourceGroupID: resp.ResourceGroup.Id,
+		ProjectID:       resp.Project.Id,
+		ParentFolderID:  parentSecretFolderID,
+	}
+}
+
+func (c OktaPAMClient) GetSecretFolder(ctx context.Context, resourceGroupID string, projectID string, id string) (*SecretFolder, error) {
```

Comment:
I think this is something we are going to change later to use SDK client.

Reply:
yeah, i will do this in a follow-up PR